### ISSUE TITLE
Doc Site: Documentation Mega Menu and main menu updates

### ISSUE THREAD

**Issue by rosibaj** [overture-stack/roadmap#46]
**Zeplin link** https://zpl.io/VOOk4o8
**Mega Menu**
Work with the Gatsby template to create the mega menu when user clicks on the Documentation tab on the Landing page.

- [x] The mega menu should show up, when the user clicks on the Documentation.
- [x] Clicking on the documentation tab again and anywhere else on the screen should be able to hide the mega menu.(see current product section mega menu on overture.bio for behaviour)  
- [x] Implement the mega menu item hover state in both mega menus (Products and Documentation)
- [x] Implement the dropshadow and bottom border in both mega menus (Products and Documentation)

<img width="1077" alt="2020-08-10_12-02-13" src="https://user-images.githubusercontent.com/18424138/89803861-62f69980-db01-11ea-950e-a6947257671b.png">

**Updates to the main header bar:**
- [x] Font size has changed to 16px
- [x] hover state now: dark blue text with light grey background and active state is magenta text with light grey background
- [x] Slack and GItHub buttons have gotten a bit smaller and using the same style. To be clear, we labeled the second button "Overture GitHub"
<img width="1080" alt="2020-07-31_1030" src="https://user-images.githubusercontent.com/18424138/89045264-3a102080-d319-11ea-9d5b-1177dab8384a.png">

**Update to the Footer:**
- [x] Add the documentation link to the footer. If there isn't room to split the "Other" category into two columns, just leave as one.
- [x] Change title of this section from "About" to "Other"
<img width="1433" alt="2020-07-31_1106" src="https://user-images.githubusercontent.com/18424138/89048538-f7047c00-d31d-11ea-86f5-00b12f188fff.png">

**Comment by gokul472** [comment]
@kcullion - pls add the zeplin link when ready.

**Comment by samrichca** [comment]
work remaining: some CSS changes

**Comment by samrichca** [comment]
https://5f63a9ed455465299805f4fe--overture.netlify.app

**Comment by samrichca** [comment]
https://github.com/overture-stack/website/pull/77

**Comment by b-f-chan** [comment]
Blocked waiting on #84, want to test with feature flag.  @samrichca new netlify also required as previous one has expired.  Will carryover to Sprint 7.

**Comment by b-f-chan** [comment]
Netlify still expired, requesting new link from @samrichca again

**Comment by b-f-chan** [comment]
Confirmed with @samrichca, now that #84 is verified and closed, a new merged netlify can/will be provided

**Comment by samrichca** [comment]
documentation is hidden, styling changes are included:

https://5f886706d161ad467a013ef0--overture.netlify.app/

documentation menu is visible in header and footer:

https://5f886764bd49c5012ac63ba5--overture.netlify.app/

**Comment by b-f-chan** [comment]
Tested at netlify links above.

Disabled Doc Menu:

- Feature flag works correctly, Doc menu NOT visible or accessible
- New active state displays correctly
- New hover state displays correctly
- Menu drop-shadow + bottom border displays correctly
- Menu titles items font size correctly reduced to 16 px
- Menu content tags/labels (e.g. Generate & Upload, Access & Download, etc) correctly widened with more spacing for text
- "Join us on Slack", "Overture GitHub" correct styling and font ("Get Started" now removed)
- "Other" section title correctly displays in footer
- "Documentation" link correctly NOT displayed in footer
- **Issue 1:** Footer links font-weight not the same as styled in the Zeplin provided
- **Issue 2:** (Case Studies, Services) in the wrong column, should be swapped with (About Us, Team Blog, Contact)

![image.png](https://images.zenhubusercontent.com/5c8bff46913aca2929d7074d/7c0dcca8-5f1b-4d1f-9a08-42969d21ab10)

Enabled Doc Menu:

- Feature flag works correctly, Doc menu is visible & accessible
- New active state displays correctly
- New hover state displays correctly
- Menu drop-shadow + bottom border displays correctly
- Menu titles items font size correctly reduced to 16 px
- Menu content tags/labels (e.g. Generate & Upload, Access & Download, etc) correctly widened with more spacing for text
- "Join us on Slack", "Overture GitHub" correct styling and font ("Get Started" now removed)
- "Other" section title correctly displays in footer
- "Documentation" link correctly displayed in footer
- (Documentation, Case Studies, Services) and (About Us, Team Blog, Contact) in correct columns
- **Issue 1:** Footer links font-weight not the same as styled in the Zeplin provided (same as disabled doc menu netlify link, see above)

**Comment by samrichca** [comment]
based on sprint planning today:

- file folders: one level: `/documentation/song/`
  - the song folder will have ALL the song docs pages with no deeper folders.
- URLs: 2 levels: `/documentation/song/installation/`
- URL/folder structure can be changed, that's just what i'm going with right now.
- left sidebar navigation: up to 4 levels
  - this can be changed, but there needs to be some limit, for graphQL and CSS purposes.
- sidebar nav headings can link to a page, or just open/close a menu.
- prev/next buttons will attempt to find the prev/next nav items in the same section (song is a section) that are links.
  - more technical explanation: flatten nav object, find prev/next sibling with file property.

**Comment by b-f-chan** [comment]
Need to close Sprint 7 so moving to Sprint 8.  Can close and open new ticket for fixes if required @samrichca, but since this didn't finish yet do not want to lose track.  Let me know your preference.

**Comment by samrichca** [comment]
from Kim

1. Let's make the font consistent for the mega-menu items and footer items: the weight you have in the megamenu of this link you sent me looks great - easier to read than the lighter footer weight. So please use the bolder weight in the footer
2. For the order of the other links, I think my mock isn't quite right. Let's go with the order we have in the header menu

left
Documentation
Case Studies
About Us
right
Services
Contact
Team Blog

**Comment by samrichca** [comment]
with documentation link

![image.png](https://images.zenhubusercontent.com/5d2e194240c06427886a21ee/d9550e8f-7f96-4a62-b1ac-86ce484a33d0)

without
![image.png](https://images.zenhubusercontent.com/5d2e194240c06427886a21ee/a21d6858-a05b-4244-be36-b7ddcaacb439)

**Comment by samrichca** [comment]
no docs
https://5f8f485c56e4490bade9fa83--overture.netlify.app/

with docs
https://5f8f482c56e4490ce0e9f79c--overture.netlify.app/

**Comment by b-f-chan** [comment]
Tested at netlify links above:

**No Docs:**

- "Documentation" correctly hidden from mega menu and footer
- Heavier font weight used in both mega menu and footer
- Order of items in two columns under "Other" in footer are in correct order

**With Docs:**

- "Documentation" correctly displayed in mega menu and footer
- Heavier font weight used in both mega menu and footer
- Order of items in two columns under "Other" in footer are in correct order

QA complete, ready to close
